src/js/components/devices/device-details/devicetwin.js

```diff
@@ -130,7 +131,13 @@ export const DeviceTwin = ({ device, getDeviceTwin, integrations, setDeviceTwin
   const [isSync, setIsSync] = useState(true);
   const editorRef = useRef(null);
 
-  const integration = integrations.find(integration => integration.provider === externalProvider.provider);
+  var integrationIds = Object.keys(device.twinsByIntegration ?? {}).filter(n => n != 'twinError');
+  if (integrationIds.length == 0) {
+    integrationIds = integrations.filter(integration => EXTERNAL_PROVIDER[integration.provider]?.deviceTwin).map(integration => integration.id);
+  }
+
+  const integration = integrations.find(integration => integration.id === integrationIds[0]);
```

Comment:
No. This won't work once we have anyone trying out how 2 integrations interact, or someone actually uses 2 integrations...

Reply:
Honestly, I'm thinking about limiting the "cloud-based" integrations (Azure/AWS) to one per tenant.

Reply:
if we change it to show up for the proper twin, it will still work if it is one per tenant, but until then this will too likely cause problems.